### PR TITLE
[BE][CI] bump `ruff` to 0.8.4

### PR DESCRIPTION
**Pull request by XuehaiPan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #143753

Changes:

1. Bump `ruff` from 0.7.4 to 0.8.4
2. Change `%`-formatted strings to f-string
3. Change arguments with the `__`-prefix to positional-only arguments with the `/` separator in function signature.

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal @ezyang @SherlockNoMad @EikanWang @wenzhe-nrv @voznesenskym @penguinwu @zhuhaozhe @blzheng @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @LucasLLC @MeetVadakkanchery @mhorowitz @pradeepfn

### CODE REVIEW

File: torch/distributed/tensor/_dtensor_spec.py [hunk 1]
```diff
@@ -68,20 +68,20 @@ def __hash__(self) -> int:
             self._hash = self._hash_impl()
         return self._hash
 
-    def __eq__(self, __o: object) -> bool:
+    def __eq__(self, other: object, /) -> bool:
```
[the remark this answers is not in the document]
Comment (XuehaiPan) [reply torch/distributed/tensor/_dtensor_spec.py:71]: It is not necessary to make this positional-only.

[the remark this answers is not in the document]
Comment (Skylion007) [reply torch/distributed/tensor/_dtensor_spec.py:71]: Which ruff rule suggestion?

[the remark this answers is not in the document]
Comment (Skylion007) [reply torch/distributed/tensor/_dtensor_spec.py:71]: Ah, I see: https://docs.astral.sh/ruff/rules/pep484-style-positional-only-parameter/

**Review by Skylion007 — APPROVED** [review]

File: torch/package/_package_pickler.py [hunk 2]
```diff
@@ -113,8 +113,8 @@ def save_global(self, obj, name=None):
                 )
             except UnicodeEncodeError:
                 raise PicklingError(
-                    "can't pickle global identifier '%s.%s' using "
-                    "pickle protocol %i" % (module, name, self.proto)  # type: ignore[attr-defined]
+                    f"can't pickle global identifier '{module}.{name}' using "
+                    f"pickle protocol {self.proto:d}"  # type: ignore[attr-defined]
                 ) from None
```
Comment (Skylion007) [inline torch/package/_package_pickler.py:118]: Still don't get why it doesn't raise the cause of the exception of here via from. From None is silly here.

Comment (Skylion007) [reply torch/package/_package_pickler.py:118]: Fixed here: https://github.com/pytorch/pytorch/pull/143761

**Pull request closed without merging**